### DISCUSSION TITLE
mongo Compass에서 이런 Error 본 적 있나요?

### DISCUSSION THREAD

**Discussion by DeukHan** [DeukHan/Study-Web#10] [Q&A]
<img width="637" alt="image" src="https://user-images.githubusercontent.com/104338516/168614381-fc4cff69-ff5d-4c3d-999d-cb3b49b70d06.png">

@kimkmin357 @dev-heeseok

**Comment by kimkmin357** [comment]
@DeukHan @dev-heeseok 
저런 에러는 보지 못했네요; 일단 Advanced Connection Options를 눌러서 Authentication탭에서 Username/Password를 입력하고 연결하는 방법도 시도해보세요

**Reply by DeukHan** [reply]
확인 감사합니다.
Authentication 옵션을 알지 못했는데 알게되었습니다.

실제 Connect 문제는 포트 문제였습니다.
TCP로 설정하였어야 했는데 Docker에서 실수로 HTTP포트로 설정했습니다.
TCP로 재 설정하니 Invalid message 에러는 사라졌습니다.

Authentication를 입력하여 진행할 때 인증 실패 에러가 발생하네요..ㅜㅜ
<img width="1544" alt="image" src="https://user-images.githubusercontent.com/104338516/168830946-e5ee4d9a-c335-4594-9e8c-ab5d53910869.png">
Docker에 환경변수가 잘 등록되지 않았나 봅니다.

**Reply by DeukHan** [reply]
Connect 버튼 누르면 에러 뜰 때 로그입니다.
<img width="273" alt="image" src="https://user-images.githubusercontent.com/104338516/168840033-0dbe04b1-adef-42ae-acd7-889784f72376.png">

**Reply by kimkmin357** [reply]
> Connect 버튼 누르면 에러 뜰 때 로그입니다.
> 
> <img width="273" alt="image" src="https://user-images.githubusercontent.com/104338516/168840033-0dbe04b1-adef-42ae-acd7-889784f72376.png">
> 
> 

마지막 내용만 보면 해당 유져가 없다는거같은데
몽고디비 컨테이너 생성시 설정했던 아이디 비번이 뭔가 잘못 설정됐을라나.. 흠

**Reply by DeukHan** [reply]
계속 오탈자 검사 했는데 큰 문제 없어보이는데...🤔
좀 더 찾아보겠습니당!
[image](https://user-images.githubusercontent.com/104338516/168927861-5d57d26a-0dc6-4f00-a603-15e851de1f81.jpeg)

**Reply by DeukHan** [reply]
해결했습니다....이유는 모르겠지만 환경변수에 등록한 user와 password가 적용되지 않아 admin으로 로그인 시 실패하고 있었는데
Docker에서 제공해주는 bash에서 수동으로 계정을 밀어넣었더니 성공했습니다.
1. Docker -> 터미널로 이동한다.
2. 터미널 -> 생성 -> bash로 생성한다.
3. mongo를 입력하여 mongodb 시작한다.
4. Use admin을 통해 DB를 admin DB로 변경한다.
5. 아래와 같이 입력하여 관리자 계정을 생성한다.
db.createUser({user: “$USERNAME”,pwd: “%PASSWORD”,roles: [ { role: “userAdminAnyDatabase”, db: “admin” } ]})
6. 관리자 계정을 추가하여 mongodb를 접속한다.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/104338516/169657596-7bf349e9-2b4e-488c-ad19-460b40267ceb.png">


### 성공..화면
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/104338516/169657423-a7a4d5ae-b346-4d12-87ea-0ff44ede199f.png">

@kimkmin357 @dev-heeseok

**Reply by kimkmin357** [reply]
오우 고생한흔적이 보이네요 동일한 문제는 이제 확실히 해결할 수 있겠습니다 ㅎ 굿굿

**Reply by DeukHan** [reply]
이제 제 Synology nas로 접속되는지 까지 해보겠습니다!

**Reply by dev-heeseok** [reply]
@DeukHan 
와우 고생했어요! 
셋팅이 거의 끝나가네요
